compiler/mir/mirpasses.nim

```diff
@@ -140,6 +160,22 @@ iterator uses(tree: MirTree, start, last: NodePosition): OpValue =
 
     dec i
 
+iterator potentialMutations(tree: MirTree, start, last: NodePosition): OpValue =
+  ## Returns in an unspecified order all values that are:
+  ## - passed to parameters that allow for mutations
+  ## - turned into pointer rvalues (i.e., have their address taken)
+  ## - have mutable views created of them
+  ## None of the above means that the used value is really mutated in the
+  ## specified range, rather it means that the value *could* be mutated.
+  var i = start
+  while i <= last:
+    # all ``mnkTag`` nodes currently imply some sort of mutation/change
```

Comment:
I'm increasingly starting to think that generalizing the original `mnkModify` into `mnkTag` was a mistake. As the comment mentions, all current value tags represent some form of mutation, but the name `mnkTag` doesn't really convey that.

I'll think about it some more, but my current opinion is that `mnkTag` should become `mnkMutation` (or similar).

Reply:
unless the intention is to generalize across other effects, I agree I think a name indicating mutation/memory makes sense.